test/e2e/resize_nodes.go

```diff
@@ -0,0 +1,331 @@
+	}
+}
+
+var _ = Describe("ResizeNodes", func() {
+	supportedProviders := []string{"gce"}
+	var testName string
+	var c *client.Client
+	var ns string
+	var testingNs *api.Namespace
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		expectNoError(err)
+		testingNs, err = createTestingNS("resize-nodes", c)
+		ns = testingNs.Name
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		if !providerIs(supportedProviders...) {
+			return
+		}
+		By(fmt.Sprintf("destroying namespace for this suite %v", ns))
+		if err := c.Namespaces().Delete(ns); err != nil {
+			Failf("Couldn't delete ns %s", err)
+		}
+		By(fmt.Sprintf("restoring the original node instance group size"))
+		if err := resizeNodeInstanceGroup(testContext.CloudConfig.NumNodes); err != nil {
+			Failf("Couldn't restore the original node instance group size: %s", err)
+		}
+		if err := waitForNodeInstanceGroupSize(testContext.CloudConfig.NumNodes); err != nil {
+			Failf("Couldn't restore the original node instance group size: %s", err)
+		}
+		if err := waitForClusterSize(c, testContext.CloudConfig.NumNodes); err != nil {
+			Failf("Couldn't restore the original cluster size: %s", err)
+		}
+	})
+
+	testName = "should be able to delete nodes."
+	It(testName, func() {
+		Logf("starting test %v", testContext.CloudConfig.NumNodes)
+
+		if !providerIs(supportedProviders...) {
+			By(fmt.Sprintf("Skipping %q test, which is only supported for providers %v (not %s)",
+				testName, supportedProviders, testContext.Provider))
+			return
+		}
+
+		if testContext.CloudConfig.NumNodes < 2 {
+			By(fmt.Sprintf("skipping %q test, which requires at lease 2 nodes (not %v)",
+				testName, testContext.CloudConfig.NumNodes))
+			return
+		}
+
+		// Create a replication controller for a service that serves its hostname.
+		// The source for the Docker containter kubernetes/serve_hostname is in contrib/for-demos/serve_hostname
+		name := "my-hostname-delete-node-" + string(util.NewUUID())
+		replicas := testContext.CloudConfig.NumNodes
+		createServeHostnameReplicationController(c, ns, name, replicas)
+		pods := waitForPodsCreated(c, ns, name, replicas)
+		waitForPodsRunning(c, pods)
+		verifyPodsListining(c, ns, name, pods)
+
+		By(fmt.Sprintf("decreasing cluster size to %v", replicas-1))
+		err := resizeNodeInstanceGroup(replicas - 1)
+		if err != nil {
+			Failf("Failed to decreas node instance group: %v", err)
+		}
+		err = waitForNodeInstanceGroupSize(replicas - 1)
+		if err != nil {
+			Failf("Failed to decreas node instance group: %v", err)
+		}
+		err = waitForClusterSize(c, replicas-1)
+		if err != nil {
+			Failf("Failed to decreas cluster size: %v", err)
+		}
+
+		By("verifying whether the pods from the removed node are recreated")
+		pods = waitForPodsCreated(c, ns, name, replicas)
+		for _, p := range pods.Items {
+			Logf("Pod: %v / %v", p.Name, p.Spec.Host)
```

Comment:
Is this debugging logging? Might want to remove this to clean up the test output.